#when function called itself, it is called recursion
